Configure and enable eslint

SLaks opened the pull request:
And fix all lint errors

**Note**: This PR continues from #124 (to set up GitHub actions and fix other errors); merge that PR first to see a simpler diff.

Or see the diff from #124 [here](https://github.com/SLaks/tikkun.io/compare/github-actions...eslint)

SLaks committed:
Project: Set up eslint in package.json and GitHub Actions

SLaks committed:
Fix all eslint errors

SLaks committed:
FIx compiler errors introduced by fixing lint errors

This catches an actual type issue.

akivajgordon commented:
@SLaks I merged #123 and #124. Do you want to check on the merge conflicts here and re-push?

SLaks committed:
Project: Add title to GitHub action

SLaks committed:
Project: Change pull_request_target to pull_request

The code in the action requires this to be `pull_request`.

This might only work for PRs from your own repo (not from forks), in which case I'll try something else.
Or it might start working on the next PR after you merge this.

SLaks committed:
Run Prettier on all source files

SLaks commented:
This is now ready.

I _think_ that the preview action will start working _after_ you merge this PR (so that the action YAML _in your repo_ grants in write access).

For now, can you force merge and ignore the failure?
